running-automation-testing.md

```diff
@@ -0,0 +1,207 @@
+
+#### First task: npm install
+
+Now that we have created the plan, we must now configure a job. 
+
+![config-job](assets/config-job.png)
+
+Click on 'Add task'. 
+
+In this guideline, we will be running the NodeJS test script, so we need to select the task type 'npm'. 
+
+![npm-task](assets/npm-task.png)
+
+In the 'Node.js executable' section, make sure the executable is pointing to the directory that has your node version. You can do this by clicking 'Add a new executable'. 
+
+![new-executable](assets/new-executable.png)
+
+For the command, put 'install'. 
+
+Click 'Save'. 
+
+![npm-install](assets/npm-install.png)
+
+#### Second task: run automation script
+
+Add another task of type 'npm'. 
+
+In the 'Node.js executable' section, make sure the executable is the same as the 'npm install' task we created previously. 
+
+For the command, run your automation script. 
+
+If necessary, you can click on 'Advanced options' and provide environment variables. 
+
+![run-task](assets/run-task.png)
+
+Click 'Save'. 
+
+### Build
+
+Click on 'Create' to start your first build. 
+
+If your build is a failure, you can check the "Logs" tab. 
+
+![failure](assets/failure.png)
+
+To see further details on the log, click on 'Default Job'. 
+
+![default-job](assets/default-job.png)
+
+To edit your plan, you can click on it near the top of the screen, next to where it says 'Build dashboard'. 
+
+![go-to-job](assets/go-to-job.png)
+
+Then click on the edit pencil icon on the right side of the plan you wish to edit. 
+
+![edit-plan](assets/edit-plan.png)
+
+Now you can select any job you want to edit. 
+
+![select-job](assets/select-job.png)
+
+Once you are done editing, you can click on the 'Run' button near the top right corner of the screen. 
+
+![run-plan](assets/run-plan.png)
+
+If your build was successful, check Kobiton cloud devices to see if a test session was created. 
+
+## Get the automation session data through Kobiton REST API
+
+- Update test result to session (https://api.kobiton.com/docs/#update-session-information)
+
+```
+PUT /sessions/{sessionId}
+``` 
+To make a request:
+
+1. Encode your credential in base64 for HTTP Basic Authentication, you may use below command and note the text result
+
+```
+echo -n <your username>:<your api-key> | base64
+```
+2. Use encoded base64 from above in curl commands, like below
+
+```
+curl -X GET https://api.kobiton.com/v1/sessions 
+  -H 'Authorization: Basic dGVzdHVzZXI6MTIzZWQtMTIzZmFjLTkxMzdkY2E='
+  -H 'Accept: application/json'
+```
+
+Below are necessary Kobiton Rest API endpoints that you may need.
+
+- Get session info
+```
+GET /sessions{sessionId}
+```
+To make a request:
+```
+curl -X GET https://api.kobiton.com/v1/sessions/{sessionId}
+  -H 'Authorization: Basic dGVzdHVzZXI6MTIzZWQtMTIzZmFjLTkxMzdkY2E='
+  -H 'Accept: application/json'
+```
+
+With this line of code, you can print the session information to the console. 
+```javascript
+console.log(sessionCapabilities)
```

Comment:
how can I use a function of javascript to get the result of a curl (bash) request ?

Reply:
I think they can check other languages as listed in my note:
> For more details on how to retrieve information about your session, go to https://api.kobiton.com/docs/

Reply:
What is the result when I run this command in bash ? What will it return and show to us ?